src/test/java/org/apache/cassandra/sidecar/metrics/FilteringMetricRegistryTest.java

```diff
@@ -0,0 +1,216 @@
+                                                                           Collections.singletonList(vertxFilter));
+        FilteringMetricRegistry metricRegistry = (FilteringMetricRegistry) registryProvider.getOrCreate();
+
+        metricRegistry.meter("sidecar.metric.exact");
+        assertThat(metricRegistry.getMetrics().containsKey("sidecar.metric.exact")).isTrue();
+        metricRegistry.timer("vertx.eventbus.message_transfer_time");
+        assertThat(metricRegistry.getIncludedMetrics().containsKey("vertx.eventbus.message_transfer_time")).isFalse();
+    }
+
+    @Test
+    void testMultipleMatchingFilterWithOneExclude()
+    {
+        MetricFilter.Equals exactFilter = new MetricFilter.Equals("sidecar.metric.exact");
+        MetricFilter.Regex regexFilter = new MetricFilter.Regex("sidecar.*");
+        MetricRegistryFactory registryFactory = new MetricRegistryFactory("cassandra_sidecar_" + UUID.randomUUID(),
+                                                                          Collections.singletonList(regexFilter),
+                                                                          Collections.singletonList(exactFilter));
+        FilteringMetricRegistry metricRegistry = (FilteringMetricRegistry) registryFactory.getOrCreate();
+
+        metricRegistry.meter("sidecar.metric.exact");
+        assertThat(metricRegistry.getIncludedMetrics().containsKey("sidecar.metric.exact")).isFalse();
+    }
+
+    @Test
+    void testExclusionsWithServer(VertxTestContext context)
+    {
+        ClassLoader classLoader = FilteringMetricRegistryTest.class.getClassLoader();
+        Path yamlPath = writeResourceToPath(classLoader, confPath, "config/sidecar_metrics.yaml");
+        Injector injector = Guice.createInjector(new MainModule(yamlPath));
+        Server server = injector.getInstance(Server.class);
+        Vertx vertx = injector.getInstance(Vertx.class);
+
+        Checkpoint serverStarted = context.checkpoint();
+        Checkpoint waitUntilCheck = context.checkpoint();
+
+        vertx.eventBus().localConsumer(SidecarServerEvents.ON_SERVER_START.address(), message -> serverStarted.flag());
+
+        server.start()
+              .onFailure(context::failNow)
+              .onSuccess(v -> {
+                  MetricRegistryFactory registryFactory = injector.getInstance(MetricRegistryFactory.class);
+                  Pattern excludedPattern = Pattern.compile("vertx.eventbus.*");
+                  FilteringMetricRegistry globalRegistry = (FilteringMetricRegistry) registryFactory.getOrCreate();
+                  assertThat(globalRegistry.getIncludedMetrics().size()).isGreaterThanOrEqualTo(1);
+                  assertThat(globalRegistry.getIncludedMetrics()
+                                           .keySet()
+                                           .stream()
+                                           .noneMatch(key -> excludedPattern.matcher(key).matches()))
+                  .isTrue();
```

Comment:
You can instead directly do `noneMatch` within the assertion.
```suggestion
                  assertThat(globalRegistry.getIncludedMetrics()
                                           .keySet()
                                           .stream())
                                           .noneMatch(key -> excludedPattern.matcher(key).matches()));
```

Reply:
I already use `noneMatch` are you suggesting removing `isTrue`?

Reply:
Addressed this